src/libpriv/rpmostree-scripts.cxx

```diff
@@ -402,11 +377,22 @@ rpmostree_run_script_in_bwrap_container (int rootfs_fd,
    */
   rpmostree_bwrap_setenv (bwrap, "SYSTEMD_OFFLINE", "1");
 
-  glnx_autofd int stdout_fd = -1;
-  glnx_autofd int stderr_fd = -1;
-  struct ChildSetupData data = { .stdin_fd = stdin_fd,
-                                 .stdout_fd = -1,
-                                 .stderr_fd = -1, };
+  /* FDs that need to be held open until we exec; they're
+   * owned by the GSubprocessLauncher instance.
+   */
+  int stdin_fd = -1;
+  int stdout_fd = -1;
+  int stderr_fd = -1;
+  int buffered_output_fd_child = -1;
+
+  if (provided_stdin_fd != -1)
+    {
+      stdin_fd = fcntl (provided_stdin_fd, F_DUPFD_CLOEXEC, 3);
+      if (stdin_fd == -1)
+        return glnx_throw_errno_prefix (error, "fcntl");
+      rpmostree_bwrap_take_stdin_fd (bwrap, stdin_fd);
```

Comment:
Incidentally, the Rust bindings for this get this right with move semantics: https://docs.rs/gio/0.9.1/gio/struct.SubprocessLauncher.html#method.take_stdout_fd - the API consumes the argument.